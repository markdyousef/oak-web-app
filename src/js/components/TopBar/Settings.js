// @flow
import React, { Component } from 'react';
import Menu from '../shared/Dropdown';
import Avatar from '../shared/Avatar';
import { signOut } from '../../utils';
import * as types from './contants';
import {
    NavRight,
    ProfileWrapper,
    Profile,
    Dropdown,
    MenuItem,
    NavContainer
} from './styles'

type DefaultProps = {
    data: Object
};

type Props = {
    data: {
        loading?: bool,
        me?: {
            gravatar: string,
            avatar: Object,
            id: string,
            name: string,
            username: string
        }
    },
    router: {
        push?: (path: string) => void,
        replace?: (path: Object) => void
    },
    logout?: () => Promise<>,
    trackEvent?: (type: string, action?: any) => void,
    setUser?: (user: Object) => void
};

type State = {
    isOpen: bool
};

export default class Settings extends Component<DefaultProps, Props, State> {
    static defaultProps = {
        data: {},
        router: {}
    };
    state = {
        isOpen: false
    }
    props: Props;
    toSettings = () => {
        const { router: { push } } = this.props;
        if (push) push('/my-settings');
    }
    componentWillReceiveProps(nextProps: Props) {
        const { data: { loading, me }, setUser } = nextProps;

        if (!loading && me && setUser) {
            setUser({
                id: me.id,
                name: me.name,
                username: me.username
            });
        }
    }
    signOut = () => {
        const { router: { replace }, logout } = this.props;
        if (logout && replace) {
            logout()
                .then(() => {
                    replace({
                        pathname: '/login'
                    });
                })
                .catch(err => console.log(err));
        }
        signOut();
    }
    onShow = () => {
        const { trackEvent } = this.props;
        const { isOpen } = this.state;

        // Analytics
        if (trackEvent) trackEvent(types.SHOW_SETTINGS, !isOpen);

        this.setState({ isOpen: !isOpen });
    }
    render() {
        const { data: { me } } = this.props;
        const { isOpen } = this.state;
        let picture;
        // prefer avatar over gravatar
        if (me && me.gravatar) picture = me.gravatar;
        if (me && me.avatar) picture = me.avatar.urlThumb64;
        return (
            <NavRight>
                <NavContainer>
                    <ProfileWrapper>
                        <Profile onClick={this.onShow}>
                            <Avatar img={picture} />
                        </Profile>
                    </ProfileWrapper>
                    {isOpen &&
                        <Dropdown style={{ right: '-6px', marginTop: '12px' }}>
                            <Menu onClose={this.onShow}>
                                <MenuItem onClick={this.toSettings}>Settings</MenuItem>
                                <MenuItem onClick={this.signOut}>Logout</MenuItem>
                            </Menu>
                        </Dropdown>
                    }
                </NavContainer>
            </NavRight>
        );
    }
}
