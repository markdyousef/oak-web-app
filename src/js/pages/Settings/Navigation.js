// @lfow
import React, { PropTypes } from 'react';
import styled from 'styled-components';
import colors from '../../styles/colors';

const Container = styled.nav`
    width: 100%;
    height: 100%;
`;

const NavGroup = styled.div`
    margin-bottom: 48px;
    & h3 {
        font-size: 20px;
        line-height: 24px;
        color: ${props => props.active ? colors.black : colors.fadedGrey};
        margin-bottom: 8px;
        font-weight: bold;
    }
    & h5 {
        font-size: 14px;
        line-height: 17px;
        color: ${props => props.active ? colors.black : colors.fadedGrey};
        margin-bottom: 12px;
    }
    & button {
        font-size: 15px;
        padding: 0;
        border: none;
        background-color: transparent;
        cursor: pointer;
        color: ${colors.green}
    }
`;

const Navigation = ({ activeSection, onSelect }:Object) => {
    return (
        <Container>
            <NavGroup active={activeSection === 0}>
                <h3>Profile</h3>
                <h5>Change your name, @username and profile pic</h5>
                <button onClick={() => onSelect(0)}>Edit profile information</button>
            </NavGroup>
            <NavGroup active={activeSection === 1}>
                <h3>Account</h3>
                <h5>Change your email address and password</h5>
                <button onClick={() => onSelect(1)}>Edit account information</button>
            </NavGroup>
            {/* <NavGroup active={activeSection === 2}>
                <h3>Cancel Account</h3>
                <h5>Chang your Full Name and/or @username.</h5>
                <button>Edit profile information</button>
            </NavGroup> */}
        </Container>
    );
};

Navigation.propTypes = {
    activeSection: PropTypes.number.isRequired,
    onSelect: PropTypes.func.isRequired
};

export default Navigation;
