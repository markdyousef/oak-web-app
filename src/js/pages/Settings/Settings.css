.container {
    width: 100%;
    display: flex;
    flex-direction: column;
}

.settings {
    display: flex;
}

.settings h1 {
    padding: 20px;
}
