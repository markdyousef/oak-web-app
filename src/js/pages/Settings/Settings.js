import React, { Component, PropTypes } from 'react';
import styled from 'styled-components';
import colors from '../../styles/colors';
import Navigation from './Navigation';
import SettingsProfile from '../../containers/SettingsProfileContainer';

const Container = styled.div`
    width: 100%;
    height: 100%;
    position: absolute;
    display: flex;
    background-color: ${colors.white};
    justify-content: center;
    flex-wrap: wrap;
    padding-top: 80px;
`;

const Divider = styled.div`
    height: 70%;
    margin: 0 80px;
    width: 1px;
    background-color: ${colors.lightGrey};
    @media (max-width: 1000px) {
        width: 0;
    }
`;

const NavContainer = styled.div`
    max-height: 580px;
    width: 100%;
    max-width: 220px;
`;

const MainContainer = styled.div`
    max-height: 580px;
    width: 100%;
    max-width: 400px;
    & h1 {
        font-size: 24px;
    }
`;

class Settings extends Component {
    static propTypes = {
        router: PropTypes.shape({
            replace: PropTypes.func
        }).isRequired
    }
    constructor() {
        super();
        this.state = {
            activeSection: 0
        };
    }
    render() {
        const { activeSection } = this.state;
        return (
            <Container>
                <NavContainer>
                    <Navigation activeSection={activeSection} onSelect={section => this.setState({ activeSection: section })} />
                </NavContainer>
                <Divider />
                <MainContainer>
                    {(activeSection === 0) &&
                        <SettingsProfile />
                    }
                </MainContainer>
            </Container>
        );
    }
}

export default Settings;
