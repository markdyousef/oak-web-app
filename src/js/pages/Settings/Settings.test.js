import React from 'react';
import renderer from 'react-test-renderer';
import Settings from './Settings';

it('renders correctly', () => {
    const component = renderer.create(
        <Settings />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
