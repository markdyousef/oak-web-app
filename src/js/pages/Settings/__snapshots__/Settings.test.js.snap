exports[`test renders correctly 1`] = `
<div
  className={undefined}>
  <div
    className={undefined}>
    <h1>
      Settings:
    </h1>
    <button
      className="bBPMRg"
      onClick={[Function]}
      type={null}>
      Sign out
    </button>
  </div>
</div>
`;
